lib/spack/spack/reporters/extract.py

```diff
@@ -178,9 +185,13 @@ def extract_test_parts(default_name, outputs):
                 # (e.g., output=str.split, error=str.split) will not have
                 # a command printed to the test log.
                 elif msg.startswith("'") and msg.endswith("'"):
-                    if part["command"]:
-                        part["command"] += "; " + msg.replace("'", "")
+                    if part:
+                        if part["command"]:
+                            part["command"] += "; " + msg.replace("'", "")
+                        else:
+                            part["command"] = msg.replace("'", "")
```

Comment:
(Comment + Refactor) We should probably give more structure to this module in a later PR, after v0.20. Here we have 3 nested conditionals, which are performing low level operations[^1], delegating to the comment above them an explanation of what is done at a high-level and why. 

I'd prefer to have, in general, classes and methods with names that evoke what is being performed at a high-level. I think we can do that when we'll remove the deprecated methods (so clean-up + refactor).

[^1]: Is a variable false-ish? Is a string starting and ending with `'`?